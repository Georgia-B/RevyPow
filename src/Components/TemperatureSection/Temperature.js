import React from 'react';

const Temperature = (props) => {
    return (
        <div key={props.title} className="temperature">
            <div>
                <h3 className="temperature__title">{props.title}</h3>
                <span>{props.subtitle}</span>
            </div>
            <div className="temperature__value">
                {props.value} <span className="temperature__unit">&deg;C</span>
            </div>
        </div>
    );
}

export default Temperature;
