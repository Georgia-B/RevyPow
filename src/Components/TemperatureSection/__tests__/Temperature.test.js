import React from 'react'
import { cleanup, render } from '@testing-library/react'
import Temperature from '../Temperature';

describe('Components | <Temperature /> with data', () => {
    afterEach(cleanup);
    it('should render correctly', () => {
        const component = render(<Temperature title="Subpeak" subtitle="1200m" value="2" />);
        expect(component.asFragment()).toMatchSnapshot();
    })
})

describe('Components | <Temperature /> with no data', () => {
    afterEach(cleanup);
    it('should render correctly', () => {
        const component = render(<Temperature />);
        expect(component.asFragment()).toMatchSnapshot();
    })
})