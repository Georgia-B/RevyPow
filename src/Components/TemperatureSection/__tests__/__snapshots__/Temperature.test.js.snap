// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components | <Temperature /> with data should render correctly 1`] = `
<DocumentFragment>
  <div
    class="temperature"
  >
    <div>
      <h3
        class="temperature__title"
      >
        Subpeak
      </h3>
      <span>
        1200m
      </span>
    </div>
    <div
      class="temperature__value"
    >
      2 
      <span
        class="temperature__unit"
      >
        Â°C
      </span>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Components | <Temperature /> with no data should render correctly 1`] = `<DocumentFragment />`;
